tests/TaskLineRenderer.test.ts

```diff
@@ -279,6 +280,50 @@ describe('task line rendering', () => {
         );
     });
 
+    it.each([
+        // Nominal cases
+        ['tag at the end #todo', 'tag at the end'],
+        ['tag #tag in the middle', 'tag in the middle'],
+        ['#now tag in the beginning', 'tag in the beginning'],
+        ['#important #s a #today #TODO lot #never of tags #now #PERFORMANCE', 'a lot of tags'],
+        ['nested tags #todo/only/today #x/y/z', 'nested tags'],
+        ['tag with hyphens #a-b-c', 'tag with hyphens'],
+
+        // Border cases
+        ['but      keep extra     spaces', 'but      keep extra     spaces'],
+    ])(
+        'should render without tags task with description "%s"',
+        async (taskDescription: string, renderedDescription: string) => {
+            await testLayoutOptions(
+                `- [ ] ${taskDescription} ⏫ 📅 2022-07-02 ⏳ 2022-07-03 🛫 2022-07-04 ➕ 2022-07-05 🔁 every day`,
+                { hideTags: true },
+                renderedDescription,
+                [' ⏫', ' 🔁 every day', ' ➕ 2022-07-05', ' 🛫 2022-07-04', ' ⏳ 2022-07-03', ' 📅 2022-07-02'],
+            );
+        },
+    );
+
+    it('"hide tags" should hide Global Filter when it is a tag despite Remove Global Filter setting', async () => {
+        const globalFilter = '#task';
+        GlobalFilter.set(globalFilter);
+
+        updateSettings({ removeGlobalFilter: false });
+        await testLayoutOptions(
+            `- [ ] ${globalFilter} Global Filter and tags are hidden #todo`,
+            { hideTags: true },
+            'Global Filter and tags are hidden', // TODO
+            [],
+        );
```

Comment:
I found these two seemingly identical test blocks to be confusing. I had to look quite hard to spot the difference between them.

Reply:
True. Shall I split into 2 tests? Or the one with `true` will be enough?

Reply:
Well I agonised over if they were both necessary - but probably giving the two tasks meaningfully different descriptions would help...

Reply:
Much better indeed, thanks! Done.